src/psfmachine/perturbation.py

```diff
@@ -0,0 +1,576 @@
+                    else:
+                        raise ValueError("Must pass other vectors in the right shape")
+            else:
+                raise ValueError("Must pass a list as other vectors")
+            self.vectors = np.hstack([self._vectors.copy(), self.other_vectors])
+        else:
+            self.vectors = self._vectors.copy()
+        if self.segments:
+            self.vectors = self._cut_segments(self.vectors)
+        # self._clean_vectors()
+        self.matrix = sparse.csr_matrix(self.bin_func(self.vectors))
+
+    def __repr__(self):
+        return "PerturbationMatrix"
+
+    @property
+    def prior_mu(self):
+        return np.zeros(self.shape[1])
+
+    @property
+    def prior_sigma(self):
+        return np.ones(self.shape[1]) * 1e4
+
+    @property
+    def breaks(self):
+        return np.where(np.diff(self.time) / np.median(np.diff(self.time)) > 5)[0] + 1
+
+    @property
+    def segment_masks(self):
+        x = np.array_split(np.arange(len(self.time)), self.breaks)
+        return np.asarray(
+            [np.in1d(np.arange(len(self.time)), x1).astype(float) for x1 in x]
+        ).T
+
+    def _cut_segments(self, vectors):
+        """
+        Cuts the data into "segments" wherever there is a break. Breaks are defined
+        as anywhere where there is a gap in data of more than 5 times the median
+        time between observations.
+        """
+        return np.hstack(
+            [
+                vectors[:, idx][:, None] * self.segment_masks
+                for idx in range(vectors.shape[1])
+            ]
+        )
+
+    def _get_focus_change(self):
+        """Finds a simple model for the focus change"""
+        focus = np.asarray(
+            [
+                np.exp(-self.focus_exptime * (self.time - self.time[b]))
+                for b in np.hstack([0, self.breaks])
+            ]
+        )
+        focus *= np.asarray(
+            [
+                ((self.time - self.time[b]) >= 0).astype(float)
+                for b in np.hstack([0, self.breaks])
+            ]
+        )
+        focus[focus < 1e-10] = 0
+        self._vectors = np.hstack([self._vectors, np.nansum(focus, axis=0)[:, None]])
+        return
+
+    # def _clean_vectors(self):
+    #     """Remove time polynomial from other vectors"""
+    #     nvec = self.poly_order + 1
+    #     if self.focus:
+    #         nvec += 1
+    #     if self.segments:
+    #         s = nvec * (len(self.breaks) + 1)
+    #     else:
+    #         s = nvec
+    #
+    #     if s != self.vectors.shape[1]:
+    #         X = self.vectors[:, :s]
+    #         w = np.linalg.solve(X.T.dot(X), X.T.dot(self.vectors[:, s:]))
+    #         self.vectors[:, s:] -= X.dot(w)
+    #         # Each segment has mean zero
+    #         self.vectors[:, s:] -= np.asarray(
+    #             [v[v != 0].mean() * (v != 0) for v in self.vectors[:, s:].T]
+    #         ).T
+    #     return
```

Comment:
maybe we can still keep this function in case we want to remove the temporal trend for other components. But it isn't necessary with the current API.